src/main/java/com/google/firebase/FirebaseApp.java

```diff
@@ -542,4 +566,32 @@ TokenRefresher create(FirebaseApp app) {
       STOPPED
     }
   }
+
+  private static FirebaseOptions getOptionsFromEnvironment() {
+    String defaultConfig = System.getenv(FIREBASE_CONFIG_ENV_VAR);
+    if (Strings.isNullOrEmpty(defaultConfig)) {
+      return new FirebaseOptions.Builder().build();
+    }
+    JsonFactory jsonFactory = Utils.getDefaultJsonFactory();
+    FirebaseOptions.Builder builder = new FirebaseOptions.Builder();
+    JsonParser parser;
+    try { 
+      if (defaultConfig.startsWith("{")) {
+        parser = jsonFactory.createJsonParser(defaultConfig);
+      } else {
+        FileReader reader;
+        reader = new FileReader(defaultConfig);
```

Comment:
Please ensure this reader gets closed. May be it happens when you call parseAndClose below. Otherwise you will need to handle it somehow.

Reply:
"Closes the parser and the underlying input stream or reader"